pkg/operator/controller/dns/controller.go

```diff
@@ -141,22 +152,22 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 				LastTransitionTime: metav1.Now(),
 			})
 			result.RequeueAfter = 30 * time.Second
+		} else if err == nil && zone.ID != "" {
```

Comment:
The `if err == nil && zone.ID != ""` can be omitted.

Reply:
If you want to check for an empty zone id in `publishRecordToZones`, you should do so before calling `r.dnsProvider.Ensure`.  You could use something like the following:
```go
		condition := iov1.DNSZoneCondition{
			Status:             string(operatorv1.ConditionUnknown),
			Type:               iov1.DNSRecordFailedConditionType,
			LastTransitionTime: metav1.Now(),
		}
		if len(zone.ID) == 0 {
			condition.Status = string(operatorv1.ConditionTrue)
			condition.Reason = "EmptyZoneId"
			condition.Message = "Zone has an empty id"
		} else if err := r.dnsProvider.Ensure(record, zone); err != nil {
			condition.Status = string(operatorv1.ConditionTrue)
			condition.Reason = "ProviderError"
			condition.Message = fmt.Sprintf("The DNS provider failed to ensure the record: %v", err)
			result.RequeueAfter = 30 * time.Second
		} else {
			condition.Status = string(operatorv1.ConditionFalse)
			condition.Reason = "ProviderSuccess"
			condition.Message = fmt.Sprintf("The DNS provider succeeded to ensure the record: %v", err)
		}
		statuses = append(statuses, iov1.DNSZoneStatus{
			DNSZone:    zone,
			Conditions: []iov1.DNSZoneCondition{condition},
		})

````

Reply:
done